### PR TITLE
Implements npmUdpdate in a separate object

### PR DESCRIPTION
**Pull request by mathieuleclaire**

### CODE REVIEW

**Comment by mathieuleclaire** [comment]
I am not sure I did what was supposed to do. I didn't expect to see all the commits history since I rebased on my repo.

**Comment by julienrf** [comment]
Well, that’s fine, I can squash the commits…

**Comment by mathieuleclaire** [comment]
Do you think it also possible to publish a minor version of scalajs-bundler so that I can use it in my application ?

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/NpmUpdateTasks.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package scalajsbundler.sbtplugin
+
+import org.scalajs.core.tools.io.{FileVirtualJSFile, RelativeVirtualFile, VirtualJSFile}
+import org.scalajs.sbtplugin.ScalaJSPlugin.AutoImport.scalaJSNativeLibraries
+import sbt.Keys.{crossTarget, streams}
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/NpmUpdateTasks.scala]: These imports are not used.

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/NpmUpdateTasks.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import sbt.{Attributed, Def, File, FileFunction, FilesInfo, IO, Task, TaskKey}
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/NpmUpdateTasks.scala]: It seems that only `sbt.IO` is used.

Comment (mathieuleclaire) [reply sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/NpmUpdateTasks.scala]: The minimal imports are:
```scala 
import org.scalajs.core.tools.io.{FileVirtualJSFile, RelativeVirtualFile, VirtualJSFile}
import scalajsbundler.{Npm, Yarn}
import sbt._
```

**Comment by julienrf** [comment]
@mathieuleclaire I’m happy to publish a new release, could help writing the changelog though?

**Comment by mathieuleclaire** [comment]
Sure, where is the doc source ?

**Comment by julienrf** [comment]
You’ll have to update the following file: https://github.com/scalacenter/scalajs-bundler/blob/master/manual/src/ornate/changelog.md

Add one entry for each merged PR with a single line description.

**Comment by mathieuleclaire** [comment]
Done for mine. I can also prepare the list of others PRs once the first changes has been done.

**Comment by mathieuleclaire** [comment]
OK, I understood how it works, I will add other PR this afternoon.

**Pull request merged**